### PR TITLE
fix: Fix llava

### PR DESCRIPTION
**Pull request by Manan17**
## Summary

Resolve a part of #723 

## Testing Done

Convergence Test:
bf16/test_mini_models:
```
python -m pytest test/convergence/bf16/test_mini_models.py -k llava
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.14, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/jobuser/Liger-Kernel
configfile: pyproject.toml
plugins: xdist-3.7.0, rerunfailures-15.1, anyio-4.9.0, lipy-config-base-32.9.0, lipy-fabric-36.1.5, lipy-test-9.1.34, datadir-1.6.1
collecting ... 
------------------------------------------------------------------------------------- live log collection -------------------------------------------------------------------------------------
INFO     datasets:config.py:54 PyTorch version 2.7.0 available.
collected 16 items / 15 deselected / 1 selected                                                                                                                                               

test/convergence/bf16/test_mini_models.py::test_mini_model[mini_llava-32-0.0001-dtype1-0.001-0.01-0.1-0.01-0.01-0.01] PASSED                                                            [100%]

```

bf16/test_mini_models_multimodal:
```
python -m pytest test/convergence/bf16/test_mini_models_multimodal.py -k llava
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.14, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/jobuser/Liger-Kernel
configfile: pyproject.toml
plugins: xdist-3.7.0, rerunfailures-15.1, anyio-4.9.0, lipy-config-base-32.9.0, lipy-fabric-36.1.5, lipy-test-9.1.34, datadir-1.6.1
collecting ... 
------------------------------------------------------------------------------------- live log collection -------------------------------------------------------------------------------------
INFO     datasets:config.py:54 PyTorch version 2.7.0 available.
collected 7 items / 6 deselected / 1 selected                                                                                                                                                 

test/convergence/bf16/test_mini_models_multimodal.py::test_mini_model_multimodal[mini_llava-32-0.0001-dtype1-0.001-0.01-0.1-0.01-0.01-0.01] PASSED
```
fp32/test_mini_models:
```
python -m pytest test/convergence/fp32/test_mini_models.py -k llava
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.14, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/jobuser/Liger-Kernel
configfile: pyproject.toml
plugins: xdist-3.7.0, rerunfailures-15.1, anyio-4.9.0, lipy-config-base-32.9.0, lipy-fabric-36.1.5, lipy-test-9.1.34, datadir-1.6.1
collecting ... 
------------------------------------------------------------------------------------- live log collection -------------------------------------------------------------------------------------
INFO     datasets:config.py:54 PyTorch version 2.7.0 available.
collected 17 items / 16 deselected / 1 selected                                                                                                                                               

test/convergence/fp32/test_mini_models.py::test_mini_model[mini_llava-32-0.0001-dtype1-1e-08-1e-05-0.005-1e-05-0.005-1e-05] PASSED                                                      [100%]
```

fp32/test_mini_models_multimodal:
```
python -m pytest test/convergence/fp32/test_mini_models_multimodal.py -k llava
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.14, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/jobuser/Liger-Kernel
configfile: pyproject.toml
plugins: xdist-3.7.0, rerunfailures-15.1, anyio-4.9.0, lipy-config-base-32.9.0, lipy-fabric-36.1.5, lipy-test-9.1.34, datadir-1.6.1
collecting ... 
------------------------------------------------------------------------------------- live log collection -------------------------------------------------------------------------------------
INFO     datasets:config.py:54 PyTorch version 2.7.0 available.
collected 7 items / 6 deselected / 1 selected                                                                                                                                                 

test/convergence/fp32/test_mini_models_multimodal.py::test_mini_model_multimodal[mini_llava-32-0.0001-dtype1-1e-08-1e-05-0.005-1e-05-0.005-1e-05] PASSED                                [100%]
```

- Hardware Type: <BLANK>
- [ ] run `make test` to ensure correctness
- [ ] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/transformers/monkey_patch.py
Comment (shimizust) [inline src/liger_kernel/transformers/monkey_patch.py:1]: From discussion with Vaibhav, we currently specify patching forward with llava_lce_forward or llava_lce_forward_deprecated based on version >= 4.49.0. Now that this is another breaking change, we can make everything <4.52.0 (?) use what was previously llava_lce_forward (move to llava_lce_forward_deprecated) and everything >= 4.52.0 to use the new llava_lce_forward. (see apply_liger_kernel_to_llava)

If users are still on transformers < 4.49.0 they will need to keep the lower liger version. We can also add a warning if transformers version < 4.49.0 that their transformers version is incompatible and they either need to upgrade transformers or downgrade liger

File: src/liger_kernel/transformers/model/llava.py [hunk 1]
```diff
@@ -292,103 +266,68 @@ def lce_forward(
...
             loss_fct = CrossEntropyLoss()
+            shift_logits = shift_logits.view(-1, self.config.vocab_size)
+            shift_labels = shift_labels.view(-1)
+            # Enable model parallelism
+            shift_labels = shift_labels.to(shift_logits.device)
+            loss = loss_fct(shift_logits, shift_labels)
 
-            flat_logits = shift_logits.view(-1, self.config.text_config.vocab_size)
-            flat_labels = shift_labels.view(-1).to(shift_logits.device)
-            loss = loss_fct(flat_logits, flat_labels)
```
Comment (Tcc0403) [inline src/liger_kernel/transformers/model/llava.py]: You only need to call `loss = self.loss_function()` with the latest transformers version.
ref: https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/llava/modeling_llava.py#L463

File: src/liger_kernel/transformers/model/llava.py [hunk 2]
```diff
@@ -292,103 +266,68 @@ def lce_forward(
...
+            lm_head_weight=self.lm_head.weight,
+            labels=labels,
+            shift_labels=shift_labels,
+            hidden_size=self.config.text_config.hidden_size,
+            **lm_kwargs,
         )
+
     else:
-        logits = self.language_model.lm_head(hidden_states)
+        logits = self.lm_head(hidden_states)
```
Comment (Tcc0403) [inline src/liger_kernel/transformers/model/llava.py]:
```suggestion
        logits = self.lm_head(kept_hidden_states)
```
